### PR TITLE
fe: support http interface that forbit use global_dict to speed up sql 

### PR DESCRIPTION
**Pull request by laotan332**
now,  the option cbo_enable_low_cardinality_optimize is set true default
it mean, if a column is varchar type and cardinality is less than 256， it will use global dict to speed up when sql execute.

if some thing go wrong or you don't want to use this options 
you can forbid it by http like
  
post   /api/global_dict/table/forbit?db_name=default_cluster:test&table_name=test_basic&enable=0

and open it like 

post   /api/global_dict/table/forbit?db_name=default_cluster:test&table_name=test_basic&enable=1

### CODE REVIEW

**Comment by imay** [comment]
@laotan332 Please make your commit title meaningful. And should explain why and what do you create this PR in the commit message. Otherwise others don't know the context of this work.

File: fe/fe-core/src/main/java/com/starrocks/journal/JournalEntity.java [hunk 1]
```diff
@@ -520,6 +520,11 @@ public void readFields(DataInput in) throws IOException {
                 isRead = true;
                 break;
             }
+            case OperationType.OP_SET_FORBIT_GLOBAL_DICT: {
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/journal/JournalEntity.java:523]: Could combine into line 518

File: fe/fe-core/src/main/java/com/starrocks/catalog/Catalog.java [hunk 2]
```diff
@@ -7096,6 +7096,41 @@ public void replayDropFunction(FunctionSearchDesc functionSearchDesc) {
         db.replayDropFunction(functionSearchDesc);
     }
 
+    public void setHasForbitGlobalDict(String dbName, String tableName) {
+        Map<String, String> emptyProperty = new HashMap<>();
+        Database db = getDb(dbName);
+        if (db == null) {
+            throw new Error("the DB " + dbName + "isn't  exist");
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/catalog/Catalog.java:7103]: throw DdlException

File: fe/fe-core/src/main/java/com/starrocks/catalog/Catalog.java [hunk 3]
```diff
@@ -7096,6 +7096,41 @@ public void replayDropFunction(FunctionSearchDesc functionSearchDesc) {
...
+            throw new Error("the DB " + dbName +  " tabet: " + tableName + "isn't  exist"); 
+        }
+
+        OlapTable olapTable = (OlapTable) table;
+        olapTable.setHasForbitGlobalDict();
+        ModifyTablePropertyOperationLog info = new ModifyTablePropertyOperationLog(db.getId(), table.getId(), emptyProperty);
+        editLog.logSetHasForbitGlobalDict(info);
+    }
+
+    public void replaySetHasFotbitGlobalDict(ModifyTablePropertyOperationLog info) {
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/catalog/Catalog.java:7117]: Could reuse `replayModifyTableProperty` ?

File: fe/fe-core/src/main/java/com/starrocks/catalog/Catalog.java [hunk 4]
```diff
@@ -7096,6 +7096,41 @@ public void replayDropFunction(FunctionSearchDesc functionSearchDesc) {
...
+        }
+
+        Table table = db.getTable(tableName);
+        if (table == null) {
+            throw new Error("the DB " + dbName +  " tabet: " + tableName + "isn't  exist"); 
+        }
+
+        OlapTable olapTable = (OlapTable) table;
+        olapTable.setHasForbitGlobalDict();
+        ModifyTablePropertyOperationLog info = new ModifyTablePropertyOperationLog(db.getId(), table.getId(), emptyProperty);
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/catalog/Catalog.java:7113]: Should we put the the bool value to the properties?  If in future, we want to allow the forbidden table global dict optimization.

File: fe/fe-core/src/main/java/com/starrocks/http/BaseRequest.java [hunk 5]
```diff
@@ -31,17 +31,23 @@
...
 import java.util.List;
 import java.util.Map;
 
+
 public class BaseRequest {
     protected ChannelHandlerContext context;
     protected HttpRequest request;
     protected Map<String, String> params = Maps.newHashMap();
 
+    private static final Logger LOG = LogManager.getLogger(BaseRequest.class);
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/http/BaseRequest.java:48]: Remove it

**Comment by kangkaisen** [comment]
run starrocks_fe_unittest

File: fe/fe-core/src/main/java/com/starrocks/catalog/Catalog.java [hunk 6]
```diff
@@ -5854,6 +5854,29 @@ public void modifyTableInMemoryMeta(Database db, OlapTable table, Map<String, St
...
+    public void setHasForbitGlobalDict(String dbName, String tableName, boolean isForbit) throws DdlException {
+        Map<String, String> property = new HashMap<>();
+        Database db = getDb(dbName);
+        if (db == null) {
+            throw new DdlException("the DB " + dbName + "isn't  exist");
+        }
+
+        Table table = db.getTable(tableName);
+        if (table == null) {
+            throw new DdlException("the DB " + dbName +  " tabet: " + tableName + "isn't  exist");
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/catalog/Catalog.java:5866]:
```suggestion
            throw new DdlException("the DB " + dbName +  " table: " + tableName + "isn't  exist"); 
```

File: fe/fe-core/src/main/java/com/starrocks/catalog/Catalog.java [hunk 7]
```diff
@@ -5854,6 +5854,29 @@ public void modifyTableInMemoryMeta(Database db, OlapTable table, Map<String, St
...
+
+        Table table = db.getTable(tableName);
+        if (table == null) {
+            throw new DdlException("the DB " + dbName +  " tabet: " + tableName + "isn't  exist"); 
+        }
+
+        OlapTable olapTable = (OlapTable) table;
+        olapTable.setHasForbitGlobalDict(isForbit);
+        if (isForbit) {
+            property.put("enable_low_card_dict", "0");
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/catalog/Catalog.java:5872]: Could define a constant value in `PropertyAnalyzer`

File: fe/fe-core/src/main/java/com/starrocks/catalog/Catalog.java [hunk 8]
```diff
@@ -5863,27 +5886,41 @@ public void replayModifyTableProperty(short opCode, ModifyTablePropertyOperation
         db.writeLock();
         try {
             OlapTable olapTable = (OlapTable) db.getTable(tableId);
-            TableProperty tableProperty = olapTable.getTableProperty();
-            if (tableProperty == null) {
-                olapTable.setTableProperty(new TableProperty(properties).buildProperty(opCode));
+            if (opCode == OperationType.OP_SET_FORBIT_GLOBAL_DICT) {
+                String enAble = properties.get("enable_low_card_dict");
+                if (enAble == null) {
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/catalog/Catalog.java:5891]: Could use `Preconditions.checkState` here. normally, we won't throw `Error` directly.

File: fe/fe-core/src/main/java/com/starrocks/catalog/Catalog.java [hunk 9]
```diff
@@ -5854,6 +5854,29 @@ public void modifyTableInMemoryMeta(Database db, OlapTable table, Map<String, St
...
+        if (db == null) {
+            throw new DdlException("the DB " + dbName + "isn't  exist");
+        }
+
+        Table table = db.getTable(tableName);
+        if (table == null) {
+            throw new DdlException("the DB " + dbName +  " table: " + tableName + "isn't  exist"); 
+        }
+
+        OlapTable olapTable = (OlapTable) table;
```
Comment (stdpain) [inline fe/fe-core/src/main/java/com/starrocks/catalog/Catalog.java:5869]:
```suggestion
        if (table instanceof OlapTable) {
            OlapTable olapTable = (OlapTable) table;
```

**Comment by kangkaisen** [comment]
run starrocks_fe_unittest

**Review by kangkaisen — APPROVED** [review]

**Review by stdpain — APPROVED** [review]

**Pull request merged**